src/frameworks/opencl/api/types.rs

```diff
@@ -272,6 +272,28 @@ pub static CL_CONTEXT_DEVICES:                           uint = 0x1081;
 pub static CL_CONTEXT_PROPERTIES:                        uint = 0x1082;
 pub static CL_CONTEXT_NUM_DEVICES:                       uint = 0x1083;
 
+enum_from_primitive! {
+    /// OpenCL context info constants.
+    #[derive(PartialEq, Debug, Copy, Clone)]
+    #[repr(C)]
+    pub enum ContextInfoQuery {
+        REFERENCE_COUNT = 0x1080,
+        DEVICES = 0x1081,
+        PROPERTIES = 0x1082,
+        NUM_DEVICES = 0x1083
+    }
+}
+
+/// OpenCL context info constants.
+#[derive(PartialEq, Debug)]
+#[repr(C)]
+pub enum ContextInfo {
+    ReferenceCount(uint),
+    NumDevices(uint),
+    ContextProperties(context_properties),
+    Devices(Vec<uint>)
+}
```

Comment:
is this better located in the layer above? or is this an API type? it feels like this should be one layer above and not part of the API :) nice modeling though